### ISSUE TITLE
`ImportDepsChecker` false positive for JDK classes missing from Android bootclasspath when using Java 17 toolchain with `aar_import`

### ISSUE THREAD

**Issue by willxiao-amzn** [bazelbuild/rules_android#495]
## Description

When using `aar_import` with a Java 17 toolchain (`remotejdk_17`), `ImportDepsChecker` reports spurious missing dependencies on standard JDK classes that are not part of the Android bootclasspath. This is a false positive — these classes are only referenced because core JDK types implement interfaces or extend classes introduced in newer Java versions, but those types are not (and should not need to be) available at runtime on Android.

## Environment

- Bazel version: 8.4.2
- rules_android version: 0.7.2
- Java toolchain: `remotejdk_17`
- OS: Linux x86_64

## Reproduction

1. Configure Java 17 in `.bazelrc`:
```
build --tool_java_language_version=17
build --tool_java_runtime_version=remotejdk_17
build --java_language_version=17
build --java_runtime_version=remotejdk_17
```

2. Have any `aar_import` target whose AAR references standard JDK types (even transitively):
```python
load("@rules_android//rules:rules.bzl", "aar_import")

aar_import(
    name = "example_lib",
    aar = "example-lib.aar",
    deps = [
        "@maven//:androidx_annotation_annotation_jvm",
        "@maven//:org_jetbrains_kotlin_kotlin_stdlib",
    ],
)
```

3. Build:
```
bazel build --config android //:example_lib
```

4. Observe errors like

```
ERROR: The dependencies for the following 1 jar(s) are not complete.
      1. bazel-out/.../classes_and_libs_merged.jar
The details are listed below:
Missing java.lang.constant.Constable
Indirectly missing class java.lang.constant.Constable. Referenced by:
    java.lang.invoke.MethodHandle
    java.lang.invoke.MethodType
```

Depending on which libraries are involved, similar failures occur for other standard JDK classes including (but not limited to):

- `sun/misc/` (e.g. `sun.misc.Unsafe`)
- `java/beans/` (e.g. `java.beans.PropertyChangeListener`)
- `org/w3c/dom/bootstrap/` (e.g. `org.w3c.dom.bootstrap.DOMImplementationRegistry`)

## Root Cause
In `rules/aar_import/impl.bzl`, `_create_import_deps_check` passes the Java toolchain's bootclasspath to `ImportDepsChecker`:

```
bootclasspath = ctx.attr._java_toolchain[java_common.JavaToolchainInfo].bootclasspath
```

The `gen_java_base_extras_jar` target in `tools/android/BUILD` extracts a subset of `platformclasspath` to supplement the Android bootclasspath for the checker:

```
run_singlejar(
    name = "gen_java_base_extras_jar",
    srcs = [
        "@rules_java//toolchains:platformclasspath",
    ],
    out = "java_base_extras.jar",
    include_prefixes = [
        "java/lang/invoke/",
        "java/lang/annotation/",
    ],
    visibility = ["//visibility:private"],
)
```

This allowlist approach is inherently fragile. With Java 17, `java.lang.invoke.MethodType` implements `java.lang.constant.Constable` (added in Java 12), so the checker follows the superinterface and fails when it can't resolve it. The same pattern applies to any JDK class that gained new supertypes or references in Java 12–17. As the JDK evolves, this list will continue to fall behind.

## Proposed Fix
Rather than maintaining an ever-growing allowlist of individual prefixes, `gen_java_base_extras_jar` should include the full `platformclasspath` (or at minimum all `java/`, `javax/`, `sun/`, and `org/` standard library packages). The checker's job is to verify that the AAR's own dependencies are declared — it should not fail because the JDK's internal class hierarchy is incomplete on the supplemented bootclasspath.

If including the full platform classpath is undesirable for jar size reasons, a more robust prefix list would at least cover the known gaps:

```
include_prefixes = [
    "java/",
    "javax/",
    "jdk/",
    "org/w3c/",
    "org/xml/",
    "sun/",
]
```
This covers standard JDK namespaces comprehensively rather than chasing individual packages as they surface.

## Workarounds
1. Disable all validations (loses other useful checks):
```
build --run_validations=false
```

2. Add affected packages to `AAR_IMPORT_DEPS_CHECKER_FALLBACK` in `rules/acls/aar_import_deps_checker.bzl` — disables the check per-package but requires ongoing maintenance (and a local fork of the repository).

Neither is ideal since the first disables unrelated validations and the second requires additions every time a new prebuilt AAR triggers the issue.

**Comment by ted-xie** [comment]
This should be fixed by #556.
